clicommand/env.go

```diff
@@ -32,14 +34,33 @@ var EnvCommand = cli.Command{
 			Usage:  "Pretty print the JSON output",
 			EnvVar: "BUILDKITE_AGENT_ENV_PRETTY",
 		},
+		cli.BoolFlag{
+			Name:  "env-file",
+			Usage: "Source environment from file described by $BUILDKITE_ENV_FILE",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Print a single environment variable by `NAME` as raw text followed by a newline",
+		},
 	},
 	Action: func(c *cli.Context) error {
-		env := os.Environ()
-		envMap := make(map[string]string, len(env))
+		var envMap map[string]string
+
+		if c.Bool("env-file") {
+			envMap = mustLoadEnvFile(os.Getenv("BUILDKITE_ENV_FILE"))
+		} else {
+			env := os.Environ()
+			envMap = make(map[string]string, len(env))
+
+			for _, e := range env {
+				k, v, _ := strings.Cut(e, "=")
+				envMap[k] = v
+			}
```

Comment:
I suspect @moskyb's efforts to also use this command in windows will cause a merge conflict here.